build/stf-run-ci/tasks/setup_stf_local_build.yml

```diff
@@ -54,23 +54,25 @@
       cmd: oc apply -f {{ base_dir }}/working/smart-gateway-operator-bundle/manifests/smart-gateway-operator.clusterserviceversion.yaml -n "{{ namespace }}"
 
 # --- Service Telemetry Operator ---
-- name: Generate Service Telemetry Operator CSV
-  ansible.builtin.shell:
-    chdir: "{{ base_dir }}"
-    cmd: |
-      LOGFILE="{{ logfile_dir }}/sto_gen_bundle.log" \
-      OPERATOR_SDK="{{ base_dir }}/working/operator-sdk-{{ operator_sdk_v0 }}" \
-      WORKING_DIR="{{ base_dir }}/working/service-telemetry-operator-bundle" \
-      RELATED_IMAGE_PROMETHEUS_WEBHOOK_SNMP={{ prometheus_webhook_snmp_image_path | parse_image | quote }} \
-      RELATED_IMAGE_PROMETHEUS_WEBHOOK_SNMP_TAG={{ prometheus_webhook_snmp_image_path | parse_tag | quote }} \
-      OPERATOR_IMAGE={{ sto_image_path | parse_image | quote }} \
-      OPERATOR_TAG={{ sto_image_path | parse_tag | quote }} \
-      ./generate_bundle.sh
-  register: generate_bundle_sto
-
-- name: Results of STO bundle generation
-  ansible.builtin.debug:
-    var: generate_bundle_sto.stdout
+- block:
+  - name: Generate Service Telemetry Operator CSV
+    ansible.builtin.shell:
+      chdir: "{{ base_dir }}"
+      cmd: |
+        LOGFILE="{{ logfile_dir }}/sto_gen_bundle.log" \
+        OPERATOR_SDK="{{ base_dir }}/working/operator-sdk-{{ operator_sdk_v0 }}" \
+        WORKING_DIR="{{ base_dir }}/working/service-telemetry-operator-bundle" \
+        RELATED_IMAGE_PROMETHEUS_WEBHOOK_SNMP={{ prometheus_webhook_snmp_image_path | parse_image | quote }} \
+        RELATED_IMAGE_PROMETHEUS_WEBHOOK_SNMP_TAG={{ prometheus_webhook_snmp_image_path | parse_tag | quote }} \
+        OPERATOR_IMAGE={{ sto_image_path | parse_image | quote }} \
+        OPERATOR_TAG={{ sto_image_path | parse_tag | quote }} \
+        ./generate_bundle.sh
+    register: generate_bundle_sto
+  rescue:
```

Comment:
Does this work? I thought if ansible.builtin.shell fails, it will return a non-zero generate_bundle_sto.rc, but otherwise I don't expect the rescue block to be called. I could be wrong and will trust your testing over my assumptions if you say this actually works.

Reply:
See the other comment, you can trust me and test it out yourself.